### PR TITLE
SALTO-4970: Salesforce fetch with changes detection for types with nested instances

### PR DESCRIPTION
**Pull request by tamtamirr**
Salesforce fetch with changes detection for types with nested instances

---

Add generic implementation for all the types that has sub instances and related types.

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/salesforce-adapter/src/adapter.ts [hunk 1]
```diff
@@ -730,6 +756,7 @@ export default class SalesforceAdapter implements AdapterOperations {
 
   private async getElemIdsByTypeFromSource(): Promise<Record<string, ElemID[]>> {
     const metadataElements = (await awu(await this.elementsSource.getAll()).toArray())
+      .filter(element => !constants.NON_LISTED_ELEMENT_IDS.includes(element.elemID.getFullName()))
```
[the remark this answers is not in the document]
Comment (tamtamirr) [reply packages/salesforce-adapter/src/adapter.ts:759]: Yes, adding the support in the new types exposed this deletion issue

File: packages/salesforce-adapter/src/adapter.ts [hunk 2]
```diff
@@ -350,12 +359,17 @@ const getMetadataTypesFromElementsSource = async (
     .toArray()
 )
 
+type CreateFiltersRunnerParams = {
+  fetchProfile: FetchProfile
+  lastChangeDateOfTypesWithNestedInstances?: LastChangeDateOfTypesWithNestedInstances
```
[the remark this answers is not in the document]
Comment (tamtamirr) [reply packages/salesforce-adapter/src/adapter.ts:364]: Changed the input as we've discussed 🙏

File: packages/salesforce-adapter/src/adapter.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tamtamirr) [reply packages/salesforce-adapter/src/adapter.ts:759]: Added comment that further explains the issue 🙏

File: packages/salesforce-adapter/src/adapter.ts [hunk 3]
```diff
@@ -350,12 +358,17 @@ const getMetadataTypesFromElementsSource = async (
     .toArray()
 )
 
+type CreateFiltersRunnerParams = {
+  fetchProfile: FetchProfile
+  contextOverrides?: Partial<FilterContext>
```
Comment (tomsellek) [inline packages/salesforce-adapter/src/adapter.ts:363]: 🔥 🔥 🔥

**Review by tomsellek — APPROVED** [review]
LGTM!

**Pull request merged**